File: solutions/2026-jan/1/resolution_streak.py
def resolution_streak(days):
    streak = 0
    for day in days:
        if day[0] >= 10000 and day[1] <= 120 and day[2] >= 5:
            streak += 1
        else:
            break

    if streak == len(days):
        return f"Resolution on track: {streak} day streak."
    return f"Resolution failed on day {streak + 1}: {streak} day streak."

File: solutions/2026-jan/1/resolution_streak.js
function resolutionStreak(days) {
    let streak = 0;
    for (const day of days) {
        if (day[0] >= 10000 && day[1] <= 120 && day[2] >= 5) streak++;
        else break;
    }
    if (streak == days.length)
        return `Resolution on track: ${streak} day streak.`;
    return `Resolution failed on day ${streak + 1}: ${streak} day streak.`;
}
